native/jni/core/module.cpp

```diff
@@ -722,6 +725,13 @@ static void collect_modules(bool open_zygisk) {
 #error Unsupported ABI
 #endif
             }
+        } else {
+            // Disable zygisk modules when zygisk is not enabled
+            if (faccessat(modfd, "zygisk", F_OK, 0) == 0) {
```

Comment:
Something like this would make me happy,  as developers that only use zygisk to enhance their modules and not as a core functionality would not be forced to ask the user to install or not to install the zygisk files on install
```cpp
            if (faccessat(modfd, "zygisk_optional", F_OK, 0) != 0 
                && accessat(modfd, "zygisk", F_OK, 0) == 0) {
```

Reply:
this pr is just for UI shows so I won't make any API changes.

Reply:
You disable every module that contain `zygisk` files whatever they need `zygisk` or not.
You are just breaking zygisk api itself at this point. 
- I'm ok with disabling modules for UI
- I'm also ok with disabling zygisk modules that require zygisk to work
- But I'm not ok with forcing modules maker to make guesses on whatever they should add or not zygisk files to their module if it's optional for them.

You are actually forcefully disabling any module containing a `zygisk` file whatever they are needed or not without providing any proper alternative for the ones needing them.

Reply:
I am ok with the last point since the same logic applies to riru modules now.

Reply:
You know what, just leave it like this, anyone can do a PR
For now it's good enough because "zygisk" is just at it's beginning stage.
I will probably wait for your PR to be accepted before doing mine on top of the changes you made.
(Or you can just apply the code I suggested, if you want)
Anyway I will probably do a PR anyway because this new behavior need also to be documented.